cmake: run target does not work on native_sim if run alone

**Describe the bug**
running `ninja run` (or `west -t run`) does not build the needed dependencies for the native simulator targets (native_sim or nrf*bsim targets).
The executable is not built, only the zephyr library.

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=native_sim ../samples/hello_world/
3. ninja run
4. See error

**Expected behavior**
Proper dependencies are built (the executable), and ninja run works

**Impact**
Confused users who do not understand why either things don't run, or are not rebuilt.

**Logs and console output**
```
$ ninja run
[1/80] Generating include/generated/version.h
-- Zephyr version: 3.5.0-rc1 (zephyr), build: v3.5.0-rc1-18-g6de57d098557
[79/80] cd zephyr/build && zephyr/build/zephyr/zephyr.exe
/bin/sh: 1: zephyr/build/zephyr/zephyr.exe: not found
FAILED: zephyr/CMakeFiles/run_native zephyr/build/zephyr/CMakeFiles/run_native 
cd zephyr/build && zephyr/build/zephyr/zephyr.exe
ninja: build stopped: subcommand failed.
```
**Environment (please complete the following information):**
 - OS: Linux (Ubuntu 22.04)
 - Toolchain (e.g Zephyr SDK, ...): Host gcc (12)
 - Commit SHA or Version used: Today's main 6de57d0985577def3aabc05b3d2d645ba882fe26

--- a/cmake/emu/native.cmake
+++ b/cmake/emu/native.cmake
@@ -3,7 +3,9 @@
 add_custom_target(run_native
   COMMAND
   ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_EXE_NAME}
-  DEPENDS ${logical_target_for_zephyr_elf}
+  DEPENDS
+    ${logical_target_for_zephyr_elf}
+    $<$<TARGET_EXISTS:native_runner_executable>:native_runner_executable>
   WORKING_DIRECTORY ${APPLICATION_BINARY_DIR}
   USES_TERMINAL
   )

